kbase-extension/static/kbase/js/widgets/narrative_core/kbaseAppCard.js

```diff
@@ -0,0 +1,145 @@
+/**
+ *  kbaseAppCard.js -- used making App cards in narrative
+ *
+ *  Authors: diane.z.zheng@gmail.com
+ *
+ *   Example and expected out put:
+ *
+ *     var $card = new kbaseDataCard(
+                {   
+                    //expected values
+                    app: app,
+                    self: self
+                    favorite: app.favorite,
+
+                    //values with default, passing in value will override default
+                    date: date,
+                    createdBy: author,
+                    type: type,
+
+                    //optional values
+                    moreContent: $moreContent,
+                });
+ *
+*/
+
+define (
+    [
+        'bootstrap',
+        'util/icon',
+        'bluebird',
+        'util/bootstrapDialog',
+        'util/display',
+        'kbase/js/widgets/narrative_core/kbaseCardLayout',
+        'narrativeConfig',
+        'jquery'
+    ], function(
+        bootstrap,
+        Icon,
+        Promise,
+        BootstrapDialog,
+        DisplayUtil,
+        kbaseCardLayout,
+        Config,
+        $
+    ) {
+        function KbaseAppCard(entry) {
+            var self = entry.self;
```

Comment:
What is 'entry.self' in this case? Another widget that has some attributes like a catalog client and several other options? This seems a little too dependent on passing in the calling object and expecting it to have things like the catalog client. Should probably just instantiate that here, as necessary.